sdk/core/corehttp/corehttp/runtime/pipeline/_base_async.py

```diff
@@ -26,17 +26,24 @@
 from __future__ import annotations
 from types import TracebackType
 from typing import Any, Union, Generic, TypeVar, List, Optional, Iterable, Type
-from typing_extensions import AsyncContextManager
+from typing_extensions import AsyncContextManager, TypeGuard
 
 from . import PipelineRequest, PipelineResponse, PipelineContext
 from ..policies import AsyncHTTPPolicy, SansIOHTTPPolicy
+from ..pipeline._base import is_sansio_http_policy
 from ._tools_async import await_result as _await_result
 from ...transport import AsyncHttpTransport
 
 AsyncHTTPResponseType = TypeVar("AsyncHTTPResponseType")
 HTTPRequestType = TypeVar("HTTPRequestType")
 
 
+def is_async_http_policy(policy) -> TypeGuard[AsyncHTTPPolicy]:
+    if hasattr(policy, "send"):
```

Comment:
Since this would also return True in the sync HTTPPolicy case, I'm wondering if we need to add a check here to ensure we do have an async HTTPPolicy, something like inspect.iscoroutinefunction(policy.send).

Reply:
updated!